### ISSUE TITLE
Adding after_failed_sign_up_path_for helper to specify which view to render.

### ISSUE THREAD

**Issue by hsyn** [plataformatec/devise#3999]
Let me explain a little bit what I am trying to do here. 
I have different registration forms depending on the selected registration types. 

instead of the single `new` template in `registrations`, I created 3 different templates to be used in order to show different kinds of ways to register as a user.

The problem is, when the form is submitted and `create` method is handling the registration, it will always render the `new` template if object is not persisted. It is implemented here --> https://github.com/plataformatec/devise/blob/3-stable/app/controllers/devise/registrations_controller.rb#L32

What I want to do is implementing a method called  `after_failed_sign_up_path_for` to specify a template to render the resource. 

What do you guys think about it? 

If you guys think that it is a good idea, we will need to replace `respond_with` with `render` instead because `respond_with` will always ignore the location parameter since it will try to render `new` if it is there.

**Comment by ulissesalmeida** [comment]
@hsyn I don't know if it is good a idea to remove `respond_with`, because it came with [more behavior](http://edgeapi.rubyonrails.org/classes/ActionController/Responder.html) and is not backward compatible.

If the problem is template rendering, I think you can do something like in your own app code:

```erb
#new.html.erb
# Render the partial depending of your user type
render signup_form_for(@user), user: user
# Your helper can print something: admin_form, user_form, awesome_user_form
```

**Comment by hsyn** [comment]
Hey @ulissesalmeida ,
Thanks for taking your time on the issue. 
Yes, template rendering problem can be solved somehow. I was mostly thinking giving flexibility to developers to choose where to send failed signups. At the same time, i totally understand that `respond_with` comes with more behavior that it is not easy to remove right now.

I customised registration controller in my app and added this.
 
`render after_failed_sign_up_path_for(resource)` 

It was just an idea, no pressure 👍 

Have a great day!

**Comment by secondstreetmedia** [comment]
I'm wondering if this is the same issue I've just run into, in a different way. I have three different user types (i.e. Devise models - `standard_user`, `admin_user`, and `super_admin_user`) and each type has a preferences page where they can change a number of values associated with their account. The forms are handled using AJAX, so only the particular attribute they are changing is refreshed, not the whole page.

I tried to handle all of these updates with a separate `preferences_controller`, but ran into an issue when tackling my `change_email` method. I'm assuming this is because I'm requiring the current password to update it as opposed to the other attribute updates I have implemented so far for which I have not required the current password (Note: I have not yet implemented `change_password`, for which, too, I will be requiring the current password).

So, the `change_email` method in my `preferences_controller`ultimately calls the Devise `resource.update` method. The update works fine if the input values are valid, but if I submit the form with blank values, what renders in response is the default Devise `registration#edit` form with the validation error messages, when I would rather, minimally, return the user to their preferences page with a general email update error message, or, ideally, return them to their preferences page with the `change_email` form open and the individual validation error messages displayed.

I'm guessing based on the discussion above, there is a work-around for me to achieve my desired implementation, but this seems like a case where I, like the OP, would be able to utilize a `after_failed_sign_up_path_for` helper to great advantage. And I'm guessing the implementation I'm attempting is likely not one that is unique to me. So, I, too, would be wondering if there is a way to provide the functionality of the requested helper without compromising the current `respond_with` functionality.

Food for thought.  Thanks!
